app/controllers/marcxml_controller.rb

```diff
@@ -1,26 +1,20 @@
 # frozen_string_literal: true
 
 class MarcxmlController < ApplicationController #:nodoc:
-  before_action :set_marcxml_resource
-
   rescue_from(SymphonyReader::ResponseError) do |e|
     render status: :internal_server_error, plain: e.message
   end
 
   def catkey
-    render plain: @marcxml.catkey
+    render plain: SymphonyReader.new(**marcxml_resource_params).fetch_catkey
```

Comment:
Can we be specific about what is getting passed here?
```suggestion
render plain: SymphonyReader.new(barcode: params.fetch(:barcode)).fetch_catkey
```

Reply:
yes, i can do something like this instead: `SymphonyReader.new(barcode: marcxml_resource_params[:barcode]).fetch_catkey`